### PR TITLE
Use latest CI images from origin-aggregated-logging

### PR DESCRIPTION
**Pull request by JAORMX**
Lets ues the latest CI images from origin-aggregated-logging in the e2e
tests.

### CODE REVIEW

**Comment by JAORMX** [comment]
@richm This is the first try... not sure about the labels though. Didn't know where to find them.

File: hack/vendor/olm-test-script/e2e-olm.sh [hunk 1]
```diff
@@ -70,6 +70,13 @@ if [ -n "${OPERATOR_IMAGE:-}" ] ; then
   CSV=$(echo "$CSV" | sed -e "s~image:.*~image: ${OPERATOR_IMAGE}\n~" | indent ApiVersion)
 fi
 
+ci_image_prefix="registry.svc.ci.openshift.org/origin/${VERSION}"
```
Comment (richm) [inline hack/vendor/olm-test-script/e2e-olm.sh]: while this should work, what the script really should do is to use `${IMAGE_FORMAT}`

you'll also need to update the elasticsearch image and the oauth-proxy image

https://github.com/openshift/origin-aggregated-logging/blob/master/hack/deploy-logging.sh#L171
and
https://github.com/openshift/origin-aggregated-logging/blob/master/hack/deploy-logging.sh#L195

**Comment by JAORMX** [comment]
@richm the elasticsearch operator image is handled elsewhere. Or did you meanthe elasticsearch image in general? cause that seems to be handled different from everything else for some reason: https://github.com/openshift/cluster-logging-operator/blob/master/hack/testing/test-367-logforwarding.sh#L59

**Comment by richm** [comment]
> @richm the elasticsearch operator image is handled elsewhere.

That will also need to use the one that `${IMAGE_FORMAT}` points at.

> Or did you meanthe elasticsearch image in general?

Yes - the elasticsearch image and the oauth-proxy image will have to come from the location that `${IMAGE_FORMAT}` points at.

> cause that seems to be handled different from everything else for some reason: https://github.com/openshift/cluster-logging-operator/blob/master/hack/testing/test-367-logforwarding.sh#L59

@jcantrill ?

**Comment by JAORMX** [comment]
@richm @jcantrill it passed CI. Is there anywhere in the logs where we can see if the correct images were used?

File: hack/vendor/olm-test-script/e2e-olm.sh [hunk 2]
```diff
@@ -70,6 +71,26 @@ if [ -n "${OPERATOR_IMAGE:-}" ] ; then
...
+
+# Use latest origin-aggregated-logging
+if [ -n "${IMAGE_FORMAT:-}" ] ; then
```
Comment (jcantrill) [inline hack/vendor/olm-test-script/e2e-olm.sh]: We need to figure out a way to pass this in. Ideally this script can be contributed back to be generic for operators and this breaks that: https://github.com/shawn-hurley/olm-test-script

**Comment by richm** [comment]
> @richm @jcantrill it passed CI. Is there anywhere in the logs where we can see if the correct images were used?

If the test dumps the deployment objects or the actual pod objects, you can see from the `image:` field there which image was used.

File: hack/vendor/olm-test-script/e2e-olm.sh (hunk 2 above)
Comment (JAORMX) [reply hack/vendor/olm-test-script/e2e-olm.sh]: Isn't this already passed though? it came from /hack/utils

**Comment by JAORMX** [comment]
> > @richm @jcantrill it passed CI. Is there anywhere in the logs where we can see if the correct images were used?
> 
> If the test dumps the deployment objects or the actual pod objects, you can see from the `image:` field there which image was used.

It dumps the pod objects, but they're not picked up since they get deleted before the artifact gathering. So we don't know :confused:

**Comment by richm** [comment]
> > > @richm @jcantrill it passed CI. Is there anywhere in the logs where we can see if the correct images were used?
> > 
> > 
> > If the test dumps the deployment objects or the actual pod objects, you can see from the `image:` field there which image was used.
> 
> It dumps the pod objects, but they're not picked up since they get deleted before the artifact gathering. So we don't know confused

. . . . which is why we need that test script enhancement that starts a background job that watches for new objects/pods, dumps the object contents, does something like `oc logs -f $pod`, etc. so we don't have to add a golang cleanup routine that does this.

File: hack/vendor/olm-test-script/e2e-olm.sh (hunk 2 above)
Comment (richm) [reply hack/vendor/olm-test-script/e2e-olm.sh]: `${IMAGE_FORMAT:-}` comes from the CI environment - so if you are trying to test outside of the CI environment, you'll have to use the alternate methods as in https://github.com/openshift/origin-aggregated-logging/blob/master/hack/deploy-logging.sh#L187-L192

File: hack/vendor/olm-test-script/e2e-olm.sh [hunk 3]
```diff
@@ -70,6 +71,26 @@ if [ -n "${OPERATOR_IMAGE:-}" ] ; then
...
+    FLUENTD_IMAGE=$(sed -e "s,\${component},logging-fluentd," <(echo $IMAGE_FORMAT))
+    KIBANA_IMAGE=$(sed -e "s,\${component},logging-kibana5," <(echo $IMAGE_FORMAT))
+    CURATOR_IMAGE=$(sed -e "s,\${component},logging-curator5," <(echo $IMAGE_FORMAT))
+    OAUTH_PROXY_IMAGE=$(sed -e "s,\${component},oauth-proxy," <(echo $IMAGE_FORMAT))
```
Comment (richm) [inline hack/vendor/olm-test-script/e2e-olm.sh]: Add `ELASTICSEARCH_IMAGE` also

File: hack/vendor/olm-test-script/e2e-olm.sh [hunk 4]
```diff
@@ -70,6 +71,26 @@ if [ -n "${OPERATOR_IMAGE:-}" ] ; then
...
+else
+    FLUENTD_IMAGE="${ci_image_prefix}:logging-fluentd"
+    KIBANA_IMAGE="${ci_image_prefix}:logging-kibana5"
+    CURATOR_IMAGE="${ci_image_prefix}:logging-curator5"
+    OAUTH_PROXY_IMAGE="${ci_image_prefix}:oauth-proxy"
```
Comment (richm) [inline hack/vendor/olm-test-script/e2e-olm.sh]: Add `ELASTICSEARCH_IMAGE` also

File: hack/vendor/olm-test-script/e2e-olm.sh [hunk 5]
```diff
@@ -70,6 +71,26 @@ if [ -n "${OPERATOR_IMAGE:-}" ] ; then
...
+fi
+
+CSV=$(echo "$CSV" | sed -e "s~quay.io/openshift/origin-logging-fluentd:latest~${FLUENTD_IMAGE}~" | indent apiVersion)
+CSV=$(echo "$CSV" | sed -e "s~quay.io/openshift/origin-logging-kibana5:latest~${KIBANA_IMAGE}~" | indent apiVersion)
+CSV=$(echo "$CSV" | sed -e "s~quay.io/openshift/origin-logging-curator5:latest~${CURATOR_IMAGE}~" | indent apiVersion)
+CSV=$(echo "$CSV" | sed -e "s~quay.io/openshift/origin-oauth-proxy:latest~${OAUTH_PROXY_IMAGE}~" | indent apiVersion)
```
Comment (richm) [inline hack/vendor/olm-test-script/e2e-olm.sh]: Add `ELASTICSEARCH_IMAGE` also

**Comment by JAORMX** [comment]
This is replaced by https://github.com/openshift/cluster-logging-operator/pull/302

**Pull request closed without merging**